src/Clapi/NamespaceTracker.hs

```diff
@@ -223,39 +231,56 @@ handleDisconnect i = do
 registerSubs
   :: (Ord i, Monad m)
   => i -> OutboundClientInitialisationDigest -> StateT (NstState i) m ()
-registerSubs i (OutboundClientDigest cas defs _tas dd _errs) = modify go
+registerSubs i (OutboundClientDigest cas postDefs defs _tas dd _errs) =
+    modify go
   where
     newDRegsForI = Set.union (Map.keysSet cas) (alKeysSet dd)
-    go (NstState owners tRegs dRegs) = NstState owners
+    go (NstState owners ptRegs tRegs dRegs) = NstState owners
+      (Map.insertWith (<>) i (Map.keysSet postDefs) ptRegs)
       (Map.insertWith (<>) i (Map.keysSet defs) tRegs)
       (Map.insertWith (<>) i newDRegsForI dRegs)
 
 subResponse :: OutboundClientInitialisationDigest -> FrcDigest
-subResponse (OutboundClientDigest cOps defs tas dd errs) =
-    FrcDigest typesInError pathsInError defs tas dd cOps errs
+subResponse (OutboundClientDigest cOps postDefs defs tas dd errs) =
+    FrcDigest postTypesInError typesInError pathsInError postDefs defs tas dd
+    cOps errs
   where
-    (pathsInError, typesInError) =
+    (pathsInError, postTypesInError, typesInError) =
         foldl collectError mempty $ Map.keys errs
-    collectError (pie, tie) ei = case ei of
-        PathError p -> (Set.insert p pie, tie)
-        TypeError t -> (pie, Set.insert t tie)
-        _ -> (pie, tie)
+    collectError (pie, ptie, tie) ei = case ei of
+        -- FIXME: err index can't differentiate between POST types and normal
```

Comment:
This fixme is out of date (because of PostTypeError)?

In the type heavy branch this was done differently, it did not work out well, this looks more manageable.